pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go

```diff
@@ -99,8 +99,18 @@ func (kcs *KafkaChannelStatus) GetCondition(t apis.ConditionType) *apis.Conditio
 }
 
 // IsReady returns true if the resource is ready overall.
-func (kcs *KafkaChannelStatus) IsReady() bool {
-	return kcs.GetConditionSet().Manage(kcs).IsHappy()
+func (kc *KafkaChannel) IsReady() bool {
+	kcs := kc.Status
+	return kcs.ObservedGeneration == kc.Generation &&
+		kcCondSet.Manage(&kcs).IsHappy()
+}
```

Comment:
dumb question (not an expert here ; ) but... do we need/want to use the locking of in GetConditionSet() ?

Reply:
@travis-minke-sap I'd actually prefer this logic:

```go
func (kc *KafkaChannel) IsReady() bool {
	kcs := kc.Status
	return kcs.ObservedGeneration == kc.Generation &&
		kcs.GetCondition(KafkaChannelConditionReady).IsTrue()
}
```

see discussion: https://github.com/knative-sandbox/eventing-kafka/pull/825/files/de8ac568c42b3cf083d8625de9b7d2921e929140#diff-0ea18f3cdf62994ad696e7f896c685979e0e9296d91f1d1cad8f247b63f104e1   and I changed afterwards

Reply:
@travis-minke-sap changed to go w/ `kc.GetConditionSet...` thanks!  Misread your comment first